asserts/snapasserts/validation_sets.go

```diff
@@ -44,6 +60,45 @@ func (e *ValidationSetsConflictError) Error() string {
 	return buf.String()
 }
 
+// ValidationSetsValidationError describes an error arising
+// from validation of snaps against ValidationSets.
+type ValidationSetsValidationError struct {
+	// snapName -> validationSetKey
+	MissingSnaps       map[string][]string
+	InvalidSnaps       map[string][]string
+	WrongRevisionSnaps map[string][]string
+	// snapName -> revision map for WrongRevisionSnaps
+	Revisions map[string]snap.Revision
```

Comment:
we could merge WrongRevisionSnaps and this by having

WrongRevisionSnaps  map[string]map[snap.Revision][]string

this means the error would even work in ValidationSets is in conflict

Reply:
Good idea, done.